ui/dashboards/src/context/TemplateVariableProvider/TemplateVariableProvider.tsx

```diff
@@ -174,6 +178,10 @@ function createTemplateVariableSrvStore({ initialVariableDefinitions = [] }: Tem
                 val = val.filter((v) => v !== ALL_VALUE);
               }
             }
+            if (queryParams) {
+              const setQueryParams = queryParams[1];
+              setQueryParams({ [getURLQueryParamName(name)]: val });
```

Comment:
Pulled your branch locally and seemed to work well! I like that our URLs are actually readable so glad we didn't go with the JSON encoded params option.

In my implementation, one downside I just noticed of getting rid of 'end=now' when 'start' is relative, is once the time range switches to absolute, the end param gets appended instead of being beside 'start' ([ex1](https://localhost:3000/projects/perses/dashboards/Demo?var-instance=%24__all&start=1666721610000&var-interval=1m&var-job=grafana%2Cnode&end=1666721700000), [ex2](https://localhost:3000/projects/perses/dashboards/DemoDashboard?start=1666717080000&var-instance=demo.do.prometheus.io%3A3000&var-interval=1m&var-job=node&end=1666717125000)). 

Definitely a nit, but I wonder if there is some way we could ensure time range params and template variable params stay together, another example is when I load a dashboard without any params, it defaulted to: `?var-instance=%24__all&start=5m&var-interval=1m`

Reply:
Hmm yeah. My guess is because when a variable loads, it only updates its value and not all the values at once. I believe we would need to group the updates together to get that functionality